docs/development/tutorial/README.md

```diff
@@ -3,6 +3,7 @@
 ## Assumptions
 
 - Some familiarity with 'git' is assumed
+- A working Java environment (a JRE or JDK)
```

Comment:
'JRE' option perhaps should be removed, a JDK would be needed to compile. A person reading this doc would be interested in checking the code out, building it and running it.

Reply:
I was confused, too, but I had only a JRE and `./gradelw` somehow managed to compile it nevertheless.
This [0] stackoverflow answer indicates that Gradle uses a library.  In another answer (which I cannot find anymore), it was said that `javax.tools` is used and I found that jar file in the `~/.gradle/` directory.

[0] https://stackoverflow.com/questions/65817520/how-do-i-know-for-sure-which-version-of-javac-gradle-used-in-a-build

Reply:
Wow, that is a bit surprising and interesting to learn.

I think we'd want to keep it simpler and just state a JDK (if it's not already implied and part of setting up an IDE), as the IDE would need a JDK. Getting up and running on CLI  is a good step, but not enough to be fully productive. Notably, a person should be able to launch a debugger, go to method declarations to trace through code, etc, in order to be fully productive, which requires an IDE to do efficiently.